src/App.jsx

```diff
@@ -13,34 +15,102 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const lengthCondition = 'length';
+const alphabetCondition = 'alphabet';
+const reverseCondition = 'reverse';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getSortedList(list, condition) {
+  const listCopy = [...list];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (condition) {
+    switch (condition) {
+      case lengthCondition:
+        return listCopy.sort((a, b) => a.length - b.length);
+      case alphabetCondition:
+        return listCopy.sort();
+      default:
+        return listCopy;
+    }
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return listCopy;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortedGoods, setSortedGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortCondition, setSortCondition] = useState('');
+
+  const handleClick = condition => {
+    if (condition === reverseCondition) {
+      setIsReversed(prev => !prev);
+    } else {
+      const newSortedGoods = getSortedList(goodsFromServer, condition);
+
+      setSortCondition(condition);
+      setSortedGoods(newSortedGoods);
+    }
+  };
+
+  const reset = () => {
+    setSortCondition('');
+    setSortedGoods(goodsFromServer);
+    setIsReversed(false);
+  };
+
+  const displayedGoods = isReversed ? [...sortedGoods].reverse() : sortedGoods;
```

Comment:
```suggestion
  const sortedGoods = getSortedList(goodsFromServer, condition);
  const displayedGoods = isReversed ? [...sortedGoods].reverse() : sortedGoods;
```
`sortedGoods` are dynamic the same as `displayedGoods` so you don't need to store them in the state